writer.go

```diff
@@ -704,8 +712,29 @@ func (w *writer) write(conn *Conn, batch []Message, resch [](chan<- error)) (ret
 
 	t0 := time.Now()
 	conn.SetWriteDeadline(time.Now().Add(w.writeTimeout))
-	if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
-		w.stats.errors.observe(1)
+	attempts := 0
+	for w.retries > attempts {
+		if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
+			w.stats.errors.observe(1)
+			switch err {
+			case DuplicateSequenceNumber:
+				// we've moved beyond the batch but havn't noticed it. Just return sucess.
+				// See https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java#L617
+				break
+			case TopicAuthorizationFailed, ClusterAuthorizationFailed, GroupAuthorizationFailed:
+				err = fmt.Errorf("auth error writing messages to %s (partition %d): %s", w.topic, w.partition, err)
+				break
+
+			default:
+				attempts++
+				err = fmt.Errorf("error writing messages to %s (partition %d): %s", w.topic, w.partition, err)
```

Comment:
you're setting the err to `error writing ...` and then potentially setting the error _again_ to `error writing ...` on line `739`

Reply:
on 739 I'm printing it, but I am printing double. 
Fixing that.